internal/csfle_util.go

```diff
@@ -0,0 +1,38 @@
+// Copyright (C) MongoDB, Inc. 2022-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package internal // import "go.mongodb.org/mongo-driver/internal"
+
+import (
+	"fmt"
+
+	"go.mongodb.org/mongo-driver/x/bsonx/bsoncore"
+)
+
+// GetEncryptedStateCollectionName returns the encrypted state collection name associated with dataCollectionName.
+func GetEncryptedStateCollectionName(efBSON bsoncore.Document, dataCollectionName string, stateCollectionSuffix string) (string, error) {
```

Comment:
What are your thoughts on using enums for `stateCollectionSuffix` rather than a `string` type?  It's a little more work and overhead but can be very helpful in documenting usage.  In this case, it would look something like this:

```go
type EncryptionMethods uint8

const (
    EllipticCurveCryptography EncryptionMethods = iota 
    // not sure what these stand for
    ESC 
    ECOC
)

func (method EncryptionMethods) suffix() string {
    switch method { 
        case EllipticCurveCryptography: 
            return "ecc"
        case ESC: 
            return "esc"
        case ECOC: 
            return "ecoc"
    }
}

// GetEncryptedStateCollectionName returns the encrypted state collection name associated with dataCollectionName.
func GetEncryptedStateCollectionName(method EncryptionMethod, efBSON bsoncore.Document, dataCollectionName string) (string, error) {
    // ...
    defaultName := "enxcol_." + dataCollectionName + "." + method.suffix()
}
```

Reply:
I like that idea. And thank you for the suggested code. Done.